### PR TITLE
Table of content

### LINKED ISSUE #1528: The `INSTALLATION.md` file needs a newly generated table of contents
**Is your feature request related to a problem? Please describe.**
1. There are many missing table of contents (TOC) entries in the INSTALLATION.md file.
1. Many of the existing TOC entries are under an inappropriate parent heading

This increases the risk of installation instructions being overlooked.

**Describe the solution you'd like**
1. Add the missing table of contents (TOC) entries in the INSTALLATION.md file.
1. Adjust the resulting TOC entries under the appropriate parent headings
1. Ideally, the auto-generation of the TOC should be automatic as part of the `git` process

**Describe alternatives you've considered**
- N/A

**Approach to be followed (optional)**
- Consider using a markdown TOC autogenerator

**Additional context**
- N/A

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Veer0x1**
**What kind of change does this PR introduce?**
This PR adds a script `generate:toc` to automatically generate table of contents for all the makrdown files in repository, for this command to work, there should we a comment of `<!-- toc -->` and `<!-- tocstop -->`. The table of content will be place under this comments. If this comments aren't present in any markdown file then there will be no change in those files.
I have placed this comments in all the neccessary markdown files. This done via a `markdow-toc `npm package. I have also added instruction on how to use this script in `CONTRIBUTING.md` file

**Issue Number:**

Fixes #1528 

**Did you add tests for your changes?**

**Snapshots/Videos:**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-api/blob/master/CONTRIBUTING.md)?**

### CODE REVIEW

**Comment by Veer0x1** [comment]
@palisadoes I have checked the schema validation is passing now

Screenshot 2023-12-30 at 6 43 06 AM

File: .husky/pre-commit [hunk 1]
```diff
@@ -18,5 +18,12 @@ npm run lint:fix
 
 npm run gen:schema
 
+# Check if any markdown files have been changed
+if git diff --cached --name-only | grep -E '\.md$'
```
Comment (palisadoes) [inline .husky/pre-commit:22]: This won't work on Windows.

**Review by palisadoes — CHANGES_REQUESTED** [review]
Take a look at this approach 
- https://github.com/PalisadoesFoundation/talawa-admin/pull/1280

**Comment by Veer0x1** [comment]
Schema test is failing again, have to raise new PR 😢

**Pull request closed without merging**